docs/v2-roadmap.rst

```diff
@@ -0,0 +1,179 @@
+v2.0 Roadmap
+============
+
+
+**📅 Release and Migration Schedule**
+-------------------------------------
+
+.. important::
+
+   We're seeking `sponsors and supporters for urllib3 v2.0 on Open Collective <https://opencollective.com/urllib3>`_.
+   There's a lot of work to be done for our small team and we want to make sure
+   development can get completed on-time while also fairly compensating contributors
+   for the additional effort required for a large release like ``v2.0``.
+
+   Additional information available within the :doc:`sponsors` section of our documentation.
+
+
+We're aiming for all ``v2.x`` features to be released in **mid-to-late 2021**.
+
+Here's what the release and migration schedule will look like leading up
+to v2.0 being released:
+
+- Development of ``v2.x`` breaking changes starts.
+- Release ``v1.26.0`` with deprecation warnings for ``v2.0.0`` breaking changes.
+  This will be the last non-patch release within the ``v1.x`` stream.
+- Release ``v2.0.0-alpha1`` once all breaking changes have been completed.
+  We'll wait for users to report issues, bugs, and unexpected
+  breakages at this stage to ensure the release ``v2.0.0`` goes smoothly.
+- Development of remaining ``v2.x`` features starts.
+- Release ``v2.0.0`` which will be identical to ``v2.0.0-alpha1``.
+- Release ``v2.1.0`` with remaining ``v2.x`` features.
+
+Deprecation warnings within ``v1.26.x`` will be opt-in by default.
```

Comment:
It does not have to be explained to users, but what deprecation warnings are we going to add? Will they be fixable in 1.26.x or is the fix just to move to 2.x? This part is still unclear to me

Reply:
Since we're proposing adding deprecations to v1.26.0 we'd raise deprecation warnings whenever something would start failing on v2.0.0 so stuff like establishing via TLS <1.2 due to default `ssl_version` setting, certificate verifying with commonName, and having a Retry object that doesn't implement whatever the new API we decide for rewriting requests (functionality won't be there, but the name will be there at least)

Maybe a DeprecationWarning for Python 2.x is appropriate, could raise it on first request?

As far as what users will "have to do" if they receive warnings is either fix the certs for services they're connecting to, in the case of TLS versions they can explicitly set an ssl_version option, upgrade Python, use a proper Retry object (this one should be a no-op for almost everyone).

Reply:
Captured this todo here: https://github.com/urllib3/urllib3/issues/1973